### ISSUE TITLE
blink_led build error: Module rgb_blink contains processes, which are not supported by JSON backend (run `proc` first).

### ISSUE THREAD

**Issue by ddribin** [tinyvision-ai-inc/UPduino-v3.0#19]
I am getting a build error running `make` in the `RTL/blink_led/` directory:

```
> make
yosys -q -p "synth_ice40 -json rgb_blink.json" rgb_blink.v
ERROR: Module rgb_blink contains processes, which are not supported by JSON backend (run `proc` first).
make: *** [rgb_blink.json] Error 1
```

I am on an Intel Mac running Big Sur (11.6.5) with `yosys` installed via [Homebrew](https://brew.sh):

```
> where yosys
/usr/local/bin/yosys
> yosys --version
Yosys 0.16 (git sha1 UNKNOWN, clang 13.0.0 -fPIC -Os)
```

Version 0.16 is the latest version, release April 5, 2022 on their [Github Page](https://github.com/YosysHQ/yosys/releases).

**Comment by ddribin** [comment]
I was able to resolve this following these instructions:

https://peter.quantr.hk/2022/03/resolved-icesugar-pro-blink-v/#page-content

Here's a patch that fixes it for me. However I'm not familiar enough with `yosys` to know why this works and if this change is appropriate in all cases.


```diff
diff --git a/RTL/blink_led/Makefile b/RTL/blink_led/Makefile
index 8cb3ff2..ec0bc52 100644
--- a/RTL/blink_led/Makefile
+++ b/RTL/blink_led/Makefile
@@ -12,7 +12,7 @@ rgb_blink.asc: rgb_blink.json ../common/upduino.pcf
 	nextpnr-ice40 --up5k --package sg48 --json rgb_blink.json --pcf ../common/upduino.pcf --asc rgb_blink.asc   # run place and route
 
 rgb_blink.json: rgb_blink.v
-	yosys -q -p "synth_ice40 -json rgb_blink.json" rgb_blink.v
+	yosys -p "read_verilog rgb_blink.v" -p "synth_ice40 -json rgb_blink.json"
 
 .PHONY: flash
 flash:
```

**Comment by vr2045** [comment]
Thanks for filing! This seems pretty safe to pull into the makefile.

**Comment by ddribin** [comment]
One oddity we discovered in Discord is that using the `yosys` from (OSS CAD Suite)[https://github.com/YosysHQ/oss-cad-suite-build] works with the unmodified `Makefile`. That version is `0.16+65`:

```
> ./bin/yosys --version
Yosys 0.16+65 (git sha1 a8cc0c393, x86_64-apple-darwin20.2-clang 10.0.0-4ubuntu1 -fPIC -Os)
```

Not sure if that means stock `0.16` is busted or if the `brew` version of `yosys` is compiled with different options.

**Comment by vr2045** [comment]
So seems like this is a yosys issue, right?
eg. should we put out a message to the user from the makefile that the
particular version is busted?

**Comment by ddribin** [comment]
I don't know enough about `yosys` to know if it's a bug. FWIW, version `0.17` was just released and is available in Homebrew, and the build issue persists.

**Comment by ddribin** [comment]
Actually... I take that back. `0.17` does seem to work. I'm not sure what I was running into yesterday:

```
> yosys --version
Yosys 0.17 (git sha1 UNKNOWN, clang 13.0.0 -fPIC -Os)
> make
yosys -q -p "synth_ice40 -json rgb_blink.json" rgb_blink.v
nextpnr-ice40 --up5k --package sg48 --json rgb_blink.json --pcf ../common/upduino.pcf --asc rgb_blink.asc   # run place and route
...
```

I guess we can close this out.

**Comment by vr2045** [comment]
Working on the newer yosys, closing issue.

**Comment by ddribin** [comment]
Thanks. For kicks, I downloaded OSS CAD Suite version `20220406`, which contained close to vanilla `0.16` and it did repro:

```
> yosys --version
Yosys 0.16+2 (git sha1 4da3f2878, x86_64-apple-darwin20.2-clang 10.0.0-4ubuntu1 -fPIC -Os)
> make
yosys -q -p "synth_ice40 -json rgb_blink.json" rgb_blink.v
ERROR: Module rgb_blink contains processes, which are not supported by JSON backend (run `proc` first).
make: *** [rgb_blink.json] Error 1
```

So yeah, definitely seems like an issue fixed between `0.16` and `0.17`.
